modules/beopBidAdapter.js

```diff
@@ -0,0 +1,144 @@
+    let fpd = config.getLegacyFpd(config.getConfig('ortb2')) || {site: {keywords: []}};
+    let gdpr = bidderRequest.gdprConsent;
+    let firstSlot = slots[0];
+
+    let payloadObject = {
+      at: new Date().toString(),
+      nid: firstSlot.nid,
+      nptnid: firstSlot.nptnid,
+      pid: firstSlot.pid,
+      url: pageUrl,
+      lang: (window.navigator.language || window.navigator.languages[0]),
+      kwds: fpd.site.keywords,
+      dbg: false,
+      slts: slots,
+      is_amp: utils.deepAccess(bidderRequest, 'referrerInfo.isAmp'),
+      tc_string: (gdpr && gdpr.gdprApplies) ? gdpr.consentString : '',
+    };
+    const payloadString = JSON.stringify(payloadObject);
+    return {
+      method: 'POST',
+      url: ENDPOINT_URL,
+      data: payloadString
+    }
+  },
+  interpretResponse: function(serverResponse, request) {
+    if (serverResponse && serverResponse.body && utils.isArray(serverResponse.body.bids) && serverResponse.body.bids.length > 0) {
+      return serverResponse.body.bids;
+    }
+    return [];
+  },
+  onTimeout: function(timeoutData) {
+    if (timeoutData === null || timeoutData === undefined || Object.keys(timeoutData).length === 0) {
+      return;
+    }
+
+    let trackingParams = buildTrackingParams(timeoutData, 'timeout', timeoutData.timeout);
+
+    utils.logWarn(BIDDER_CODE + ': timed out request');
+    utils.triggerPixel(utils.buildUrl({
+      protocol: 'https',
+      hostname: 't.beop.io',
+      pathname: '/bid',
+      search: trackingParams
+    }));
+  },
+  onBidWon: function(bid) {
+    if (bid === null || bid === undefined || Object.keys(bid).length === 0) {
+      return;
+    }
+    let trackingParams = buildTrackingParams(bid, 'won', bid.cpm);
+
+    utils.logInfo(BIDDER_CODE + ': won request');
+    utils.triggerPixel(utils.buildUrl({
+      protocol: 'https',
+      hostname: 't.beop.io',
+      pathname: '/bid',
+      search: trackingParams
+    }));
+  },
+  onSetTargeting: function(bid) {}
+}
+
+function buildTrackingParams(data, info, value) {
+  return {
+    pid: data.params.accountId,
+    nid: data.params.networkId,
+    nptnid: data.params.networkPatnerId,
+    bid: data.bidId,
+    sl_n: data.adUnitCode,
+    aid: data.auctionId,
+    se_ca: 'bid',
+    se_ac: info,
+    se_va: value
+  };
+}
+
+function beOpRequestSlotsMaker(bid) {
+  const bannerReq = utils.deepAccess(bid, 'mediaTypes.banner');
+  const bannerSizes = bannerReq.sizes;
+  const publisherCurrency = utils.getValue(bid.params, 'currency') || 'EUR';
+  let floor;
+  if (typeof bid.getFloor === 'function') {
+    const floorInfo = bid.getFloor({currency: publisherCurrency, mediaType: 'banner', size: [1, 1]});
+    if (typeof floorInfo === 'object' && floorInfo.currency === publisherCurrency && !isNaN(parseFloat(floorInfo.floor))) {
+      floor = parseFloat(floorInfo.floor);
+    }
+  }
+  return {
+    sizes: utils.isArray(bannerSizes) ? utils.isArray(bannerSizes) : bid.sizes,
```

Comment:
The `isArray` call is repeated. If `bannerSizes` is an array, the value of this property should be `true`?

Reply:
No it should not, thanks